### ISSUE TITLE
Out of bounds access

### ISSUE THREAD

**Issue by urosidoki** [itoral/gl-terrain-demo#4]
https://github.com/itoral/gl-terrain-demo/blob/efbe5e77268f1c111ab6bc52a45a86d8ebad9655/src/ter-terrain.cpp#L209

I think here you can have other out of bound access. You iterate over the whole loop but inside calculate_normals you do a + 1.

I found these issues because I use your code as reference for a terrain system in my own game engine :)

Cheers,

**Comment by itoral** [comment]
Yes, I guess we would probably want to workaround this like we do for x=0 and z= 0, but on the other side of the spectrum.
